pkg/controller/backup_controller.go

```diff
@@ -425,7 +420,14 @@ func (controller *backupController) runBackup(backup *api.Backup, bucket string)
 		controller.logger.WithError(err).Error("error closing gzippedLogFile")
 	}
 
-	if err := storage.UploadBackup(log, objectStore, bucket, backup.Name, backupJSONToUpload, backupFileToUpload, logFile); err != nil {
+	// TODO get location with name = backup.StorageLocation from lister
+	var location *api.BackupStorageLocation
+	accessor, err := storage.BackupAccessor(backup.Name, location, pluginManager, controller.logger)
+	if err != nil {
+		return err
+	}
+
+	if err := accessor.SaveBackup(backupJSONToUpload, backupFileToUpload, logFile); err != nil {
```

Comment:
This doesn't easily allow something like a git backend (git wouldn't use/have `backupFileToUpload`). Take a look at https://github.com/heptio/ark/commit/c07bec22ec59ccbe68ec56e2d249072b6e5d5521 for comparison. I don't think this commit is entirely correct, but maybe we can find a way to merge our two approaches into something awsome 😄